CMSIS/NN/Source/SoftmaxFunctions/arm_softmax_s8.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2010-2019 Arm Limited or its affiliates. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_softmax_s8.c
+ * Description:  S8 softmax function
+ *
+ * $Date:        October 2019
+ * $Revision:    V.1.0.0
+ *
+ * Target Processor:  Cortex-M cores
+ *
+ * -------------------------------------------------------------------- */
+
+#include "arm_nnfunctions.h"
+
+#define ACCUM_BITS 12
+
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup Softmax
+ * @{
+ */
+void arm_softmax_s8(const int8_t *input,
+                    const uint16_t num_rows,
+                    const uint16_t row_size,
+                    const int32_t mult,
+                    const int32_t shift,
+                    const int32_t diff_min,
+                    int8_t *output)
+{
+    const int32_t mask = (1 << shift);
+
+    uint16_t row = 0;
+    uint16_t col = 0;
+    for(row = 0; row < num_rows; ++row)
```

Comment:
I think we can optimize this for loop with something like:

    const int32_t max_row_idx = row_size * num_rows;
    int32_t col = 0;
    int32_t row_idx;

    for(row_idx = 0; row_idx < max_row_idx; row_idx += row_size)
    {
        // Find the maximum value in order to ensure numerical stability
        // and initialize the output with -128
        int8_t max = input[row_idx];
        
        output[row_idx + col] = -128;
        for (col = 1; col < row_size; ++col)
        {
            max = MAX(max, input[row_idx + col]);
            
            // Initialize the output with -128
            output[row_idx + col] = -128;
        }

if you initialize the output with -128 here, you do not need the else condition at line 88